server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -1312,16 +1314,25 @@ public void onJobSubmit(final SubmitJobRequest request) {
         } 
     }
 
-    private JobDefinition insertAutoResubmitLable(JobDefinition resolvedJobDefn) {
+    private JobDefinition insertAutoResubmitLabel(JobDefinition resolvedJobDefn) {
         List<Label> labels = resolvedJobDefn.getLabels();
-        List<Label> updatedLabels = new ArrayList<>(labels);
-        updatedLabels.add(new Label("_mantis.isResubmit","true"));
-        try {
-            JobDefinition updatedJobDefn = new JobDefinition.Builder().from(resolvedJobDefn).withLabels(updatedLabels).build();
-            logger.info("Added isResubmit label");
-            return updatedJobDefn;
-        } catch (InvalidJobException e) {
-            logger.error(e.getMessage());
+
+        boolean alreadyHasResubmitLabel = labels.stream().anyMatch(label -> label.getName().equals(MANTIS_IS_RESUBMIT));
+
+        if(!alreadyHasResubmitLabel) {
+            List<Label> updatedLabels = new ArrayList<>(labels);
+            updatedLabels.add(new Label(MANTIS_IS_RESUBMIT, "true"));
+            try {
+                JobDefinition updatedJobDefn = new JobDefinition.Builder().from(resolvedJobDefn)
+                        .withLabels(updatedLabels).build();
+                logger.info("Added isResubmit label");
```

Comment:
log `jobCluster`/`jobId` with log statement, also some of the logging could be at DEBUG level in this method.